static/app/components/modals/inviteMembersModal/index.tsx

```diff
@@ -70,7 +73,28 @@ function InviteMembersModal({
         onSendInvites={sendInvites}
       >
         {({sendInvites: _sendInvites, canSend, headerInfo}) => {
-          return (
+          return organization.features.includes('invite-members-new-modal') ? (
+            <InviteMembersModalNew
+              canSend={canSend}
+              Header={Header}
+              Body={Body}
+              complete={complete}
+              Footer={Footer}
+              headerInfo={headerInfo}
+              invites={invites}
+              inviteStatus={inviteStatus}
+              member={memberResult.data}
+              pendingInvites={pendingInvites}
+              reset={reset}
+              sendingInvites={sendingInvites}
+              sendInvites={sendInvites}
+              setEmails={setEmails}
+              setRole={setRole}
+              setTeams={setTeams}
+              willInvite={willInvite}
+              error={error}
```

Comment:
Rather than passing everything into this component as a property, could we create a context to store all of these functions and variables? That way any component inside of the `InviteMembersModalNew` could access them without passing it from this level all the way down (this is an anti-pattern called [Prop Drilling](https://www.freecodecamp.org/news/prop-drilling-in-react-explained-with-examples/)).

If we used a react context to store these functions / states, the interface to this modal could be limited to what the modal cares about.

Next, we could look at the properties of a modal, for example `Header` and `Footer`. Those items in particular are more like children of the "Modal" component, and can be composable items that are namespaced to the modal. (Here's an example from a popular ui library: https://www.radix-ui.com/primitives/docs/components/dropdown-menu and here's a blog post that dives into what a namespaced component is: https://medium.com/@kunukn_95852/react-components-with-namespace-f3d169feaf91) 

Another way we could design this interface is:
```jsx
<InviteModalContext.Provider
  value={{
    canSend={canSend}
    invites={invites}
    inviteStatus={inviteStatus}
    {/* copy all the props */}
  }}
>
  <InviteMembersModalNew>
     <InviteMembersModalNew.Header>{Header}</InviteMembersModalNew.Header>
        {/*
            what would be extra cool is to pull the
            content of the modal to here, passing it as the
            'children' variable to the component. so you can
            see all the rows etc, if you did that you
            may not even need to use a context
            provider 🤔
         */}
     <InviteMembersModalNew.Footer>{Footer}</InviteMembersModalNew.Footer>
  </InviteMembersModalNew>
</InviteModalContext.Provider>
```
^^ it's looking a lot more like HTML! that makes it a lot easier for us to understand what's happening in the component at a visual level. "I have a modal, with this header, this content, and this footer" - then inside of the modal, it will understand how to open / close (for now we can leave the invite stuff to the internals of the modal since it's specifically the invite members modal).

----

Thiiiiiisss is also a fairly large refactor. so only take on this feedback if you have the time -- happy to help with anything if you do decide to take on this endeavor though!